logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ClearConversationContent.kt

```diff
@@ -0,0 +1,83 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.benasher44.uuid.uuid4
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.asset.AssetRepository
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.user.AssetId
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.feature.message.MessageSender
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.fold
+import kotlinx.datetime.Clock
+
+internal class ClearConversationContent(
+    private val conversationRepository: ConversationRepository,
+    private val assetRepository: AssetRepository
+) {
+
+    suspend operator fun invoke(conversationId: ConversationId): Either<CoreFailure, Unit> {
+        return conversationRepository.getAssetMessages(conversationId).flatMap { conversationAssetMessages ->
+            conversationAssetMessages.forEach { message ->
+                val messageContent: MessageContent = message.content
+
+                if (messageContent is MessageContent.Asset) {
+                    with(messageContent.value.remoteData) {
+                        assetRepository.deleteAsset(
+                            AssetId(
+                                assetId,
+                                assetDomain.orEmpty()
+                            ), assetToken
+                        )
+                    }
```

Comment:
This deletes the asset remotely. This means that other users won't be able to download the assets you sent after you clear the conversation.

I don't think this is the current behaviour, is it?

Maybe `assetRepository.deleteAssetLocally` is what we want?

Reply:
Isn't that something we want ?

Reply:
changed to deleteAssetLocally